p2p/net/conn/listen.go

```diff
@@ -60,57 +86,8 @@ func (l *listener) SetAddrFilters(fs *filter.Filters) {
 // Accept waits for and returns the next connection to the listener.
 // Note that unfortunately this
 func (l *listener) Accept() (net.Conn, error) {
-
-	// listeners dont have contexts. given changes dont make sense here anymore
-	// note that the parent of listener will Close, which will interrupt all io.
-	// Contexts and io don't mix.
-	ctx := context.Background()
-
-	var catcher tec.TempErrCatcher
-
-	catcher.IsTemp = func(e error) bool {
-		// ignore connection breakages up to this point. but log them
-		if e == io.EOF {
-			log.Debugf("listener ignoring conn with EOF: %s", e)
-			return true
-		}
-
-		te, ok := e.(tec.Temporary)
-		if ok {
-			log.Debugf("listener ignoring conn with temporary err: %s", e)
-			return te.Temporary()
-		}
-		return false
-	}
-
-	for {
-		maconn, err := l.Listener.Accept()
-		if err != nil {
-			if catcher.IsTemporary(err) {
-				continue
-			}
-			return nil, err
-		}
-
-		log.Debugf("listener %s got connection: %s <---> %s", l, maconn.LocalMultiaddr(), maconn.RemoteMultiaddr())
-
-		if l.filters != nil && l.filters.AddrBlocked(maconn.RemoteMultiaddr()) {
-			log.Debugf("blocked connection from %s", maconn.RemoteMultiaddr())
-			maconn.Close()
-			continue
-		}
-		// If we have a wrapper func, wrap this conn
-		if l.wrapper != nil {
-			maconn = l.wrapper(maconn)
-		}
-
-		_, _, err = l.mux.Negotiate(maconn)
-		if err != nil {
-			log.Info("negotiation of crypto protocol failed: ", err)
-			continue
-		}
-
-		c, err := newSingleConn(ctx, l.local, "", maconn)
+	for con := range l.incoming {
```

Comment:
this hides errors. it would be very hard to debug. `net.Accept` errors if there is a problem with an incoming dial, so the client of `c, err := l.Accept()` gets the feedback easily.

fix this with:

```go
type connErr struct {
  conn net.Conn
  err     error
}

for ce := range l.incoming {
  if ce.err != nil {
    return nil, ce.err
  }
  con := ce.conn
  ...
}

// in handleIncoming:
maconn, err := l.Listener.Accept()
if err != nil {
	if catcher.IsTemporary(err) {
		continue
	}

  // send the error to the client of l.Accept()
  l.incoming<- connErr{nil, err}
  return
}
```